services/api/src/models/user.js

```diff
@@ -16,6 +16,15 @@ schema.virtual('password').set(function (password) {
   this._password = password;
 });
 
+schema.methods.addAuthTokenId = function (authTokenId) {
+  this.authTokenIds.push(authTokenId);
+  // pushes out old token that most likely expired
+  // we dont want this collection to keep growing + we dont have users to have that many valid tokens
+  if (this.authTokenIds.length > 10) {
```

Comment:
Would it not maybe make more sense now to have something like:

```js
user.authTokens = [
  {
    jti: '...',
    exp: '...',
]
```

We could use the `exp` claim to expire the token and remove it. Also could allow more claims to be added if needed in the future.

Reply:
It's just that "more than 10" seems so arbitrary and there's no way to know whether they are actually expired or old. Maybe it doesn't matter but it seems like if we allow multiple devices we'll eventually hit the limit of not being logged into more than 10 devices at the same time but without any error message or way of knowing what's going on.

One other thing aside from above, I'm thinking the below line should be `= this.authTokenIds.slice(-10)` no?

Reply:
that is very good point re error messaging, lets me add the extra logic.
I suspect that means that i will have to let go of the to attempt to enforce how many active tokens a user can have.
And perhaps thats fine.